core/modules/replica/Messenger.cc

```diff
@@ -0,0 +1,89 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/Messenger.h"
+
+// System headers
+#include <stdexcept>
+
+// Qserv headers
+#include "lsst/log/Log.h"
+#include "replica/Configuration.h"
+#include "replica/ServiceProvider.h"
+
+namespace {
+
+LOG_LOGGER _log = LOG_GET("lsst.qserv.replica.Messenger");
+
+} /// namespace
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+Messenger::pointer Messenger::create(ServiceProvider::pointer const& serviceProvider,
+                                     boost::asio::io_service& io_service) {
+    return Messenger::pointer(
+        new Messenger(serviceProvider,
+                      io_service));
```

Comment:
use `make_shared`

Reply:
It's not possible due to a private constructor. And I want to have it private because this class inherits from std::enable_shared_from_this which requires to seal constructors to prevent objects being used as values or via plain-vanilla C/C++ style raw pointer.
This is BOOST ASIO pattern. You'll see it all around their code examples.